test/common/ssl/ssl_socket_test.cc

```diff
@@ -54,19 +54,118 @@ namespace Ssl {
 
 namespace {
 
-// TODO replace the long parameter list with an options object
-void testUtil(const std::string& client_ctx_yaml, const std::string& server_ctx_yaml,
-              const std::string& expected_digest, const std::string& expected_uri,
-              const std::string& expected_local_uri, const std::string& expected_serial_number,
-              const std::string& expected_subject, const std::string& expected_local_subject,
-              const std::string& expected_peer_cert, const std::string& expected_stats,
-              bool expect_success, const Network::Address::IpVersion version) {
+/**
+ * A base class to hold the options for testUtil() and testUtilV2().
+ */
+class TestUtilOptionsBase {
+public:
+  const std::string& expectedClientCertURI() const { return expected_client_cert_uri_; }
+  const std::string& expectedServerStats() const { return expected_server_stats_; }
+  bool expectSuccess() const { return expect_success_; }
+  Network::Address::IpVersion version() const { return version_; }
+
+protected:
+  TestUtilOptionsBase(const std::string& expected_server_stats, bool expect_success,
+                      Network::Address::IpVersion version)
+      : expected_server_stats_(expected_server_stats), expect_success_(expect_success),
+        version_(version) {}
+
+  void expectedClientCertURI(const std::string& expected_client_cert_uri) {
```

Comment:
Nit: can you prefix all these mutators with `set`, e.g. `setExpectedClientCertUri`? I think this will improve readability a bit.